modules/cnf-provisioning-real-time-and-low-latency-workloads.adoc

```diff
@@ -0,0 +1,242 @@
+https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html#security-context-constraints-creating_configuring-internal-oauth[Creating Security Context Constraints].
+
+[NOTE]
+====
+This procedure is fully supported with bare metal installations using RHCOS systems.
+====
+
+Establishing the right performance expectations refers to the fact that the realtime kernel is not a panacea.
+Its objective is consistent, low-latency determinism offering predictable response times.
+There is some additional kernel overhead associated with the real-time kernel.
+This is due primarily to handling hardware interrupts in separately scheduled threads.
+The increased overhead in some workloads results in some degradation in overall throughput.
+The exact amount is very workload dependent, ranging from 0% to 30%. However, it is the cost of determinism.
+
+== Provisioning a worker with real-time capabilities
+
+. Install Performance Addon Operator to the cluster.
+See https://docs.openshift.com/container-platform-ocp/4.5/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html#installing-the-performance-addon-operator_cnf-master[Installing the Performance Addon Operator].
+. (Optional) Add a node to the OCP cluster. (link to existing OCP docs)
+. (Optional) Create a new MachineConfigPool for RT nodes.
+. Add the node to the proper MachineConfigPool, using node role labels.
++
+You need to decide which nodes will be configured with real-time workloads.
+It could be all of the nodes in the cluster or a subset of the nodes.
+The Performance Addon Operator expects all of the nodes are part of a dedicated MachineConfigPool.
+If you use all of the nodes, you just point the Performance Addon Operator to the worker node role label.
+If you use a subset, you need to group the nodes into a new MachineConfigPool.
+
+. Create the PerformanceProfile with the proper set of housekeeping cores and `realTimeKernel: enabled: true`.
+
+. Specify a node selector in the PerformanceProfile, as shown here:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+ name: example-performanceprofile
+spec:
+ [...]
+  realTimeKernel:
+    enabled: true
+  nodeSelector:
+   node-role.kubernetes.io/worker-rt: ""
+----
+
+. Verify that a matching MachineConfigPool exists with a label:
++
+----
+machineconfiguration.openshift.io/role=worker-rt
+----
+
+. OpenShift will start configuring the nodes, which may involve multiple reboots. Wait for the nodes to settle.
+This can take a long time depending on the specific hardware you use. We observed total time of about 20 minutes per node.
+
+. Verify everything is working as expected.
+
+== Changing to a real-time kernel
+
+Use this command to verify that the real-time kernel is installed:
+
+----
+oc get node -o wide
+----
+
+Note the worker with the role `worker-rt` that contains the string `4.18.0-211.rt5.23.el8.x86_64`:
```

Comment:
Note the KERNEL-VERSION column of the worker with the role `worker-rt` that contains the string `rt`: